web-components/src/components/combobox/ComboBox.ts

```diff
@@ -1324,86 +1441,97 @@ export namespace ComboBox {
                   part="combobox-options"
                   role="listbox"
                   aria-label=${this.label}
-                  style=${styleMap({
-                    display: this.expanded
-                      ? this.options.length &&
-                        this.filteredOptions.length === 0 &&
-                        this.inputValue &&
-                        this.allowCustomValue
-                        ? "none"
-                        : "block"
-                      : "none",
-                    "z-index": "1"
-                  })}
+                  style=${this.checkForVirtualScroll()
+                    ? styleMap({
+                        visibility: this.expanded
+                          ? this.options.length && this.filteredOptions.length === 0 && this.inputValue
+                            ? "hidden"
+                            : "visible"
+                          : "hidden",
+                        "z-index": "1"
+                      })
+                    : styleMap({
+                        display: this.expanded
+                          ? this.options.length &&
+                            this.filteredOptions.length === 0 &&
+                            this.inputValue &&
+                            this.allowCustomValue
+                            ? "none"
+                            : "block"
+                          : "none",
+                        "z-index": "1"
+                      })}
```

Comment:
Can we simplify the nested ternary statement?

Reply:
Fixed